scripts/upgrade-hooks/1.1.0.ts

```diff
@@ -0,0 +1,251 @@
+  }
+`;
+
+const FNFT_RANGE_QUERY = `
+  query {
+    fnftranges {
+      bosonOfferId
+      startingId
+      quantity
+    }
+  }
+`;
+
+interface FNFTRange {
+  bosonOfferId: string;
+  startingId: string;
+  quantity: string;
+}
+
+export interface OfferData {
+  offerId: string;
+  itemQuantity: string;
+  firstTokenId: string;
+}
+
+/**
+ * Fetch GraphQL data for FNFT ranges.
+ */
+async function fetchFNFTRangeData(client: any): Promise<FNFTRange[]> {
+  const result = await client.query(FNFT_RANGE_QUERY, {}).toPromise();
+
+  if (result.error) {
+    throw new Error(`GraphQL error: ${result.error.message}`);
+  }
+
+  if (!result.data?.fnftranges) {
+    throw new Error("No FNFT range data found in GraphQL response");
+  }
+
+  return result.data.fnftranges;
+}
+
+/**
+ * Prepare backfill data for offers.
+ */
+export async function prepareOfferBackfillData(client: any): Promise<OfferData[]> {
+  const fnftRanges = await fetchFNFTRangeData(client);
+  const offerDataList: OfferData[] = [];
+
+  for (const range of fnftRanges) {
+    offerDataList.push({
+      offerId: range.bosonOfferId,
+      itemQuantity: range.quantity,
+      firstTokenId: range.startingId,
+    });
+  }
+
+  console.log("Prepared Offer Backfill Data:", JSON.stringify(offerDataList, null, 2));
+
+  return offerDataList;
+}
+
+/**
+ * Fetch GraphQL data for offers and tokens in specific states.
+ */
+async function fetchGraphQLData(client: any): Promise<Offer[]> {
+  const result = await client.query(OFFER_QUERY, {}).toPromise();
+
+  if (result.error) {
+    throw new Error(`GraphQL error: ${result.error.message}`);
+  }
+
+  if (!result.data?.offers) {
+    throw new Error("No data found in GraphQL response");
+  }
+
+  return result.data.offers;
+}
+
+/**
+ * Calculate fees for backfilling based on offer details.
+ */
+function calculateFees(verifierFee: bigint, facilitatorFeePercentBps: bigint, tokenId: string, price: bigint): FeeData {
+  const HUNDRED_PERCENT_BPS = 100_00n;
+  const BOSON_PROTOCOL_FEE_BPS = 50n; // 0.5%
+  const bosonProtocolFee = (price * BOSON_PROTOCOL_FEE_BPS) / HUNDRED_PERCENT_BPS;
+  const fermionFeeAmount = 0n;
+  const remainder = price - bosonProtocolFee - fermionFeeAmount - verifierFee;
+  const facilitatorFeeAmount = remainder > 0n ? (remainder * facilitatorFeePercentBps) / HUNDRED_PERCENT_BPS : 0n;
+
+  return {
+    tokenId,
+    bosonProtocolFee,
+    fermionFeeAmount,
+    verifierFee,
+    facilitatorFeeAmount,
+  };
+}
+
+/**
+ * Prepare backfill data from fetched offers.
+ */
+export async function prepareFeeBackfillData(client: any): Promise<FeeData[]> {
+  const offers = await fetchGraphQLData(client);
+  const feeDataList: FeeData[] = [];
+
+  for (const offer of offers) {
+    const verifierFee = BigInt(offer.verifierFee);
+    const facilitatorFeePercentBps = BigInt(offer.facilitatorFeePercent);
+
+    for (const token of offer.fNFTs) {
+      const price = token.priceLog?.amount ? BigInt(token.priceLog.amount) : 0n;
```

Comment:
Need a confirmation - is this the full price, or the price without the boson fee? I'm not sure what is stored in the subgraph. If it's already stored without fees, then `calculateFees` calculation are not correct.  
Please document it in either case.

Reply:
At the moment (until [this issue](https://github.com/fermionprotocol/core-components/issues/129) will be dealt with), the `priceLog.amount` in subgraph data model is defined from the the `amount` property of the `FundsEncumbered` event emitted by Boson protocol in the same transaction when the `VerificationInitiated` event is raised by Fermion).
I think it means the `actualPrice = balanceAfter - balanceBefore` as set by the price discovery client (https://github.com/bosonprotocol/boson-protocol-contracts/blob/f099648a49045c8bad1a1bdb8e2c4ffcf67b69da/contracts/protocol/clients/priceDiscovery/BosonPriceDiscovery.sol#L192)

Reply:
Ok, so this is the full price (already stripped from any external fees, such as opensea fee), which is ok.

This means that `calculateFees` in `1.0.0.ts` works ok and it looks to me, we are filling in the correct values to backfilling faucet.

I think that resolving https://github.com/fermionprotocol/core-components/issues/129 would actually make the data inconsistent if `ItemPriceObserved` would be used directly for `priceLog.amount`. Since `ItemPriceObserved` in `1.0.1` does not include the BPFee, while `1.1.0` will again contain the price with BPFee. We need to be careful how to handle this.